client/state/posts/selectors.js

```diff
@@ -90,11 +97,17 @@ export const getSitePost = createSelector(
  * @return {?Array}         Posts for the post query
  */
 export function getSitePostsForQuery( state, siteId, query ) {
-	if ( ! state.posts.queries[ siteId ] ) {
+	const manager = state.posts.queries[ siteId ];
+	if ( ! manager ) {
+		return null;
+	}
+
+	const posts = manager.getItems( query );
+	if ( ! posts ) {
 		return null;
 	}
 
-	return state.posts.queries[ siteId ].getItems( query );
+	return posts.map( ( post ) => getNormalizedPost( state, post.global_ID ) );
```

Comment:
In case you're curious, some pseudo-code for the path refactor idea:

```js
import get from 'lodash/get';

function getPost( state, globalId ) {
	const path = state.items[ globalId ];
	if ( ! path ) {
		return null;
	}

	const [ siteId, postId ] = path;
	const manager = state.queries[ siteId ];
	if ( ! manager ) {
		return null;
	}

	return manager.getItem( postId );
}

const post = getPost( {
	items: {
		'3d097cb7c5473c169bba0eb8e3c6cb64': [ 2916284, 841 ]
	},
	queries: {
		2916284: new PostQueryManager( {
			items: {
				841: { ID: 841, site_ID: 2916284, global_ID: '3d097cb7c5473c169bba0eb8e3c6cb64', title: 'Hello World' }
			},
			queries: {}
		} )
	}
}, '3d097cb7c5473c169bba0eb8e3c6cb64' );
```

Going back to your original question, if we were to adopt this, it'd be even more awkward to get items from the query manager, then map them over `getPost`, which simply looks them up again in the query manager. At that point, I'd probably agree it'd make sense to pull the normalize flow out of `getNormalizedPost` and allow it to be called directly from the `map` here.